### ISSUE TITLE
Drop dependency on func command line tool

### ISSUE THREAD

**Issue by WoodenAndrew** [microsoft/azure-maven-plugins#1678]
### Plugin name and version
azure-functions-maven-plugin 1.13
### Plugin configuration in your `pom.xml`
Irrelevant
### Expected behavior
package and run goals do not require .net functions sdk and everything can be done without any additional tools
### Actual behavior
```
[INFO] Step 8 of 8: Installing function extensions if needed
Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  extensions -> C:\Users\User\IdeaProjects\azure-function-runtime\target\azure-functions\functionapp\bin\extensions.dll
```
### Steps to reproduce the problem
Just install the .net function sdk for windows.

Looking at the source code, azure-functions-maven plugin just maps parameters to `func` tool, which is outright wrong for any java based project. In its current state the plugin does not even permit adding additional command line parameters, nor logs the commandline used to run the tool, forcing the user to monitor the running processes using `get-wmiobject -filter "name = 'func.exe'"` powershell scriptlet to even guess what the plugin is doing and mimic that while providing the **missing** parameters yourself.

In short, the plugin must be self contained.

**Comment by WoodenAndrew** [comment]
Ping @Dragas

**Comment by andxu** [comment]
function core tools are written in .net and it is required for running functions locally. so I think .net function sdk is required, could you let me know what parameter you can add to 'func host start' to avoid the requirement for .net functions sdk?

**Comment by WoodenAndrew** [comment]
What? How is that a requirement when a new JVM is spawned to run the function application?

**Comment by andxu** [comment]
Hi @WoodenAndrew , the function core tools repository is located [here](https://github.com/Azure/azure-functions-core-tools) (managed by Azure Functions team), it must adopt one language since it support multiple languages, personally I don't think it is right to have multiple distributions of function core tools using different languages since this is an endless work.
